sale_promotion/models/product_campaign.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2025 Manuel Regidor <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ProductCampaign(models.Model):
+    _name = "product.campaign"
+    _description = "Promotion"
+
+    name = fields.Char(required=True)
+    active = fields.Boolean(default=True)
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        default=lambda self: self.env.company,
+    )
+    product_campaign_item_ids = fields.One2many(
+        comodel_name="product.campaign.item",
+        inverse_name="product_campaign_id",
+        string="Items",
+    )
+
+    def _get_applicable_rules_domain(self, products, date, **kwargs):
+        self.ensure_one()
+        if products._name == "product.template":
+            templates_domain = ("product_tmpl_id", "in", products.ids)
+            products_domain = ("product_id.product_tmpl_id", "in", products.ids)
```

Comment:
Are you sure this does work in a domain?

Reply:
It does